src/types/offer.types.ts

```diff
@@ -0,0 +1,47 @@
+export type OfferConfigurationDTO = {
+  description: string;
+  dailyPrice: number;
+  selectedOptions: Record<string, string[]>;
+}
+
+export type OfferConfigurationWithId = OfferConfigurationDTO & {
+  id: number;
+}
+
+export type AvailabilityDTO = {
+  availableFrom: string;
+  availableTo: string;
+}
+
+export type SetAvailabilityDTO = {
+  offerIds: number[];
+  availabilityRanges: AvailabilityDTO[];
+}
+
+export type OfferDTO = {
+  description: string;
+  animal: {
+    animalType: string;
+  };
+  offerConfigurations: OfferConfigurationDTO[];
+  animalAmenities: string[];
+}
+
+export type OfferDTOWithId = Omit<OfferDTO, "offerConfigurations"> & {
+  id: number;
+  offerConfigurations: OfferConfigurationWithId[];
+  availabilities: AvailabilityDTO[];
+}
+
+export type AnimalSex = "MALE" | "SHE";
```

Comment:
Tutaj chyba powinno być "FEMALE" zamiast "SHE"

Reply:
Właśnie to zależy, bo np. słownik podpowiada, że **SHE**:
![image](https://github.com/user-attachments/assets/9ce94217-48df-4a25-bf96-4d048f916a36)

Ale to bez znaczenia, bo i tak docelowo wszystkie parametry zwierząt i udogodnienia będą pobierane z backendu.